src/JSON/AST.dfy

```diff
@@ -0,0 +1,18 @@
+// RUN: %verify "%s"
+
+module {:options "-functionSyntax:4"} JSON.AST {
```

Comment:
Correction, not sure `JSON.AST.JSON` is ideal for the core datatype, could we move the contents of this module into `JSON.API`?

Reply:
Since I wasn't able to move the definitions around without circular dependencies, I went with `JSON.Values.JSON` instead. The RFC itself uses "JSON Values" to define the abstract semantics in Section 3 so it seems even better than introducing "Node".